logging/cloud-client/src/main/java/com/example/logging/ListLogs.java

```diff
@@ -39,13 +39,16 @@ public static void main(String... args) throws Exception {
 
       String logFilter = "logName=projects/" + options.getProjectId() + "/logs/" + logName;
 
-      // List log entries
+      // List all log entries
       Page<LogEntry> entries = logging.listLogEntries(
           EntryListOption.filter(logFilter));
-      for (LogEntry logEntry : entries.iterateAll()) {
-        System.out.println(logEntry);
-      }
-      // Use entries.getNextPage() to paginate
+      do {
+        for (LogEntry logEntry : entries.iterateAll()) {
+          System.out.println(logEntry);
+        }
+        entries = entries.getNextPage();
+      } while(entries != null);
```

Comment:
There are forms of getting entries that aren't fully paged, but are infinite.  I'm presuming you've verified that this is actually paged?